.github/workflows/pull-request.yaml

```diff
@@ -0,0 +1,313 @@
+        sudo rm -rf $(which hd)
+        sudo mv hd /usr/local/bin
+        sudo hd install openshift/source-to-image v1.3.1
+        sudo apt-get install libkrb5-dev -y
+
+        sudo npm -g --unsafe-perm=true --allow-root install fish-pepper --build-from-source
+    - name: Test
+      run: |
+        ./test.sh
+
+  BuildCentOSJava11:
+    runs-on: ubuntu-20.04
+    steps:
+    - uses: actions/checkout@v2
+    - name: Docker meta for KubeSphere
+      id: meta
+      if: github.repository_owner == 'kubesphere'
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          kubespheredev/s2i-java-11
+          ghcr.io/${{ github.repository_owner }}/s2i-java-11
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha
+    - name: Docker meta for Contributors
+      id: metaContributors
+      if: github.repository_owner != 'kubesphere'
+      uses: docker/metadata-action@v3
+      with:
+        images: |
+          ghcr.io/${{ github.repository_owner }}/s2i-java-11
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v1
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v1
+    - name: Login to DockerHub
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
+      with:
+        username: ${{ secrets.DOCKER_HUB_USER }}
+        password: ${{ secrets.DOCKER_HUB_SECRETS }}
+    - name: Login to GHCR
+      if: github.event_name != 'pull_request'
+      uses: docker/login-action@v1
+      with:
+        registry: ghcr.io
+        username: ${{ github.repository_owner }}
+        password: ${{ secrets.GHCR_TOKEN }}
+    - name: Build and push Docker images
+      uses: docker/build-push-action@v2.4.0
+      if: github.repository_owner == 'kubesphere'
+      with:
+        context: java/images/centos-java11
+        tags: ${{ steps.meta.outputs.tags }}
+        push: ${{ github.event_name != 'pull_request' }}
+        labels: ${{ steps.meta.outputs.labels }}
+        platforms: linux/amd64,linux/arm64
+    - name: Build and push Docker images for Contributors
+      uses: docker/build-push-action@v2.4.0
+      if: github.repository_owner != 'kubesphere'
+      with:
+        context: java/images/centos-java11
+        tags: ${{ steps.metaContributors.outputs.tags }}
+        push: ${{ github.event_name != 'pull_request' }}
+        labels: ${{ steps.metaContributors.outputs.labels }}
+        platforms: linux/amd64,linux/arm64
+
+  BuildCentOS:
```

Comment:
BuildCentOs is ambiguity, and I suggest BuildCentOsJava8 is clearer.

Reply:
Excellent suggestion.